src/async_read.rs

```diff
@@ -59,3 +59,81 @@ where
         }
     }
 }
+
+#[cfg(test)]
+mod tests {
+    use super::*;
+    use crate::shutdown::ShutdownSignal;
+    use crate::test_util::temp_file;
+    use futures::FutureExt;
+    use tokio::fs::{remove_file, File};
+    use tokio::io::{AsyncBufReadExt, AsyncWriteExt, BufReader, BufWriter};
+
+    #[tokio::test]
+    async fn test_read_line_without_shutdown() {
```

Comment:
👍 thanks for these tests! I think we should leave them.

Reply:
Ok, cool